### PR TITLE
Add FreeBSD support

### PR DESCRIPTION
**Pull request by smuth4**
Now for the fruits of my previous labors! I've tested this both in the single-VM setup, and with a FreeBSD client subscribed to a Debian server. I tried to follow existing patterns, but let me know if I missed anything.

### CODE REVIEW

**Comment by cmacrae** [comment]
Wow... thanks so much @smuth4 - this is great! Let me look through the changes and do some testing

File: defaults/main.yml [hunk 1]
```diff
@@ -46,11 +46,13 @@ sensu_remote_plugins: ~
 
 # Sensu/RabbitMQ SSL certificate properties
 sensu_ssl_gen_certs: true
-sensu_ssl_client_cert: "{{ dynamic_data_store }}/{{ groups['sensu_masters'][0] }}/{{ sensu_config_path }}/ssl_generation/sensu_ssl_tool/client/cert.pem"
-sensu_ssl_client_key: "{{ dynamic_data_store }}/{{ groups['sensu_masters'][0] }}/{{ sensu_config_path }}/ssl_generation/sensu_ssl_tool/client/key.pem"
-sensu_ssl_server_cacert: "{{ dynamic_data_store }}/{{ groups['sensu_masters'][0] }}/{{ sensu_config_path }}/ssl_generation/sensu_ssl_tool/sensu_ca/cacert.pem"
-sensu_ssl_server_cert: "{{ dynamic_data_store }}/{{ groups['sensu_masters'][0] }}/{{ sensu_config_path }}/ssl_generation/sensu_ssl_tool/server/cert.pem"
-sensu_ssl_server_key: "{{ dynamic_data_store }}/{{ groups['sensu_masters'][0] }}/{{ sensu_config_path }}/ssl_generation/sensu_ssl_tool/server/key.pem"
+sensu_master_config_path: "{{ hostvars[groups['sensu_masters'][0]]['sensu_config_path'] | default('/etc/sensu') }}"
```
Comment (cmacrae) [inline defaults/main.yml:49]: Just wondering if the `default` filter is necessary here. The `sensu_config_path` is getting picked up from `vars`.
Definitely a good idea to chuck `sensu_ssl_tool_base_path` in there though :+1:

As I want this released out ASAP, I'm gunna merge this in and change that one value to remove the filter :) But, please; by all means, go ahead and put forward any conflicting opinions!

Comment (smuth4) [inline defaults/main.yml:49]: This is happening for two reasons:
A) If sensu_config_path is different on the master and client (i.e. Debian master with `/etc/sensu`, FreeBSD client with `/usr/local/etc/sensu` like I have), the file lookup will fail:

```
TASK [cmacrae.sensu : Deploy the Sensu client SSL cert/key] ********************
failed: [freebsd] => (item=/home/smuth4/dynamic-data/debian/usr/local/etc/sensu/ssl_generation/sensu_ssl_tool/client/cert.pem) => {"failed": true, "item": "/home/smuth4/dynamic-data/debian/usr/local/etc/sensu/ssl_generation/sensu_ssl_tool/client/cert.pem", "msg": "could not find src=/home/smuth4/dynamic-data/debian/usr/local/etc/sensu/ssl_generation/sensu_ssl_tool/client/cert.pem"}
failed: [freebsd] => (item=/home/smuth4/dynamic-data/debian/usr/local/etc/sensu/ssl_generation/sensu_ssl_tool/client/key.pem) => {"failed": true, "item": "/home/smuth4/dynamic-data/debian/usr/local/etc/sensu/ssl_generation/sensu_ssl_tool/client/key.pem", "msg": "could not find src=/home/smuth4/dynamic-data/debian/usr/local/etc/sensu/ssl_generation/sensu_ssl_tool/client/key.pem"}
ok: [debian] => (item=/home/smuth4/dynamic-data/debian/etc/sensu/ssl_generation/sensu_ssl_tool/client/cert.pem)
ok: [debian] => (item=/home/smuth4/dynamic-data/debian/etc/sensu/ssl_generation/sensu_ssl_tool/client/key.pem)
```

I'm getting that on the latest master branch
B) `host_vars` doesn't contain default variables, only ones that have been overriden, so `hostvars[groups['sensu_masters'][0]]['sensu_config_path']` is actually undefined for a Debian host. See https://github.com/ansible/ansible/issues/6189. I tried the `defaults` dictionary mentioned, but couldn't get it working for me.

**Comment by cmacrae** [comment]
@smuth4 - great, thanks for the explanation. I didn't know such lookups wouldn't pick up other inventory/role defined vairables.

I've re-implemented this and documented it here: http://rtfm.cmacr.ae/differing_master_environment/

**Pull request merged**